libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -145,7 +126,12 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
     }
 
     _LIBCUDACXX_INLINE_VISIBILITY
-    friend void init(barrier * __b, _CUDA_VSTD::ptrdiff_t __expected, _CUDA_VSTD::__empty_completion __completion = _CUDA_VSTD::__empty_completion()) {
+    friend void init(barrier * __b, _CUDA_VSTD::ptrdiff_t __expected) {
+        init(__b, __expected, _CUDA_VSTD::__empty_completion());
+    }
+
+    _LIBCUDACXX_INLINE_VISIBILITY
+    friend void init(barrier * __b, _CUDA_VSTD::ptrdiff_t __expected, _CUDA_VSTD::__empty_completion __completion) {
```

Comment:
No, because of complicated overload resolution rules. I needed to split this, because it became a template. You'll see that the init for the primary template of `cuda::barrier` also does this, and I don't remember exactly which rule governs this, just that I immediately remembered this happened before when I got stupid overload resolution errors this time around too.

Reply:
:sad trombone noises: